automatic/kubernetes-cli/update.ps1

```diff
@@ -25,25 +25,46 @@ function global:au_SearchReplace {
 }
 
 function global:au_GetLatest {
+    # Only report the supported Kubernetes streams.
+    # NB Upstream Kubernetes only supports the last three minor versions.
+    #    However, they can also do pre-releases before a final version is
+    #    released, so we have to check for an extra changelogs/minor
+    #    version to be able to find all the supported streams.
+    # See https://github.com/kubernetes/sig-release/blob/master/release-engineering/versioning.md
+    $minor_versions_to_keep = 3
+
     $changelog_page = Invoke-WebRequest -Uri $changelogs -UseBasicParsing
-    #Only the latest three minor versions are supported: https://github.com/kubernetes/sig-release/blob/master/release-engineering/versioning.md
-    #However, there are prereleases before the version is released, so the latest four changelogs/minor versions need to be checked.
-    $minor_version_changelogs = $changelog_page.links | ? href -match "CHANGELOG-[\d\.]+\.md" | Select-Object -Expand href -First 4
-    
+    $minor_version_changelogs = $changelog_page.links `
+      | Where-Object href -match "CHANGELOG-(?<version>\d+\.\d+)\.md`$" `
+      | Select-Object -Expand href -First ($minor_versions_to_keep+1)
```

Comment:
Any reason why you restrict the amount of links?

If there are no reason, this can be changed to
```suggestion
      | Select-Object -Expand href
```

So we can monitor all links (In case there are a patch being created for an older release at some point).

Reply:
The restrictions came from the existing package. I just made them more explicit. From what I understood, this package intent was to track only the latest version of the latest three kubernetes releases.

But now it seems you want to track all versions and releases? Even no longer supported versions? So scape all the versions that are available?

Reply:
Typically the restriction is only in place for the beginning, to not overwhelm Chocolatey Community repository with too many versions at the same time, as well as not needing to fix a lot of package versions if they fail the verifier/validator.

Most likely, this restriction was just forgotten to be removed. However, since it already has the restriction we can leave it here for this PR.